From 5f5b7cadbefe75ab8f4453f5443a99cbf716ee80 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Thu, 30 Dec 2021 07:47:28 +0200
Subject: [PATCH] [SW-68384] gaudi2: security: set axprot for psoc smi

By default the permission level of smi mstr if is
privilege therefore any transaction coming via smi
is treated as privilege transaction hence setting the
permission level to user level.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: Iec56a43a04a2ccd6f60cc0afbea87f0769994942
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 6a210d5..27ea79d 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -46,6 +46,7 @@
 #include <pmmu_hbw_mstr_if_core_hbw_regs.h>
 #include <jt_mstr_if_core_hbw_regs.h>
 #include <i2c_s_mstr_if_core_hbw_regs.h>
+#include <smi_mstr_if_core_hbw_regs.h>
 #include "common.h"
 
 static struct hl_sec_rrs_cfg hl_sec_lbw_rrs_tbl[] = {
@@ -1145,6 +1146,8 @@
 	regmap_writel(HL_SEC_AXPROT_USER_VAL, mmJT_MSTR_IF_CORE_HBW_AXPROT_OVRD);
 	/* Set i2c_s to user level */
 	regmap_writel(HL_SEC_AXPROT_USER_VAL, mmI2C_S_MSTR_IF_CORE_HBW_AXPROT_OVRD);
+	/* Set smi to user level */
+	regmap_writel(HL_SEC_AXPROT_USER_VAL, mmSMI_MSTR_IF_CORE_HBW_AXPROT_OVRD);
 }
 
 void preboot_security_init(void)
